create_and_send_customized_greeting_cards_using_openai/service.bal

```diff
@@ -18,28 +19,32 @@ type GreetingDetails record {|
 |};
 
 service / on new http:Listener(8080) {
-    resource function post greetingCard(@http:Payload GreetingDetails req) returns error? {
-        string occasion = req.occasion;
-        string specialNotes = req.specialNotes ?: "";
+    isolated resource function post greetingCard(@http:Payload GreetingDetails req) returns error? {
+        final string occasion = req.occasion;
+        final string specialNotes = req.specialNotes ?: "";
 
         fork {
             // Generate greeting text and design in parallel
-            worker greetingWorker returns string|error? {
-                text:CreateCompletionRequest textPrompt = {
-                    prompt: string `Generate a greeting for a/an ${occasion}.${"\n"}Special notes: ${specialNotes}`,
-                    model: "text-davinci-003",
-                    max_tokens: 100
-                };
-                text:CreateCompletionResponse completionRes = check openAIText->/completions.post(textPrompt);
-                return completionRes.choices[0].text;
-            }
-            worker imageWorker returns string|error? {
-                images:CreateImageRequest imagePrompt = {
-                    prompt: string `Greeting card design for ${occasion}, ${specialNotes}`
-                };
-                images:ImagesResponse imageRes = check openAIImages->/images/generations.post(imagePrompt);
-                return imageRes.data[0].url;
-            }
+                worker greetingWorker returns string|error? {
+                    chat:CreateChatCompletionRequest request = {
+                        model: "gpt-4o-mini",
+                        messages: [{
+                            "role": "user",
+                            "content": string `Generate a greeting for a/an ${occasion}.${"\n"}Special notes: ${specialNotes}`
+                            }]
+                    };
+
+                    chat:CreateChatCompletionResponse response = check openAIChat->/chat/completions.post(request);
+                    return response.choices[0].message.content;
+                }
+
+                worker imageWorker returns string|error? {
+                    images:CreateImageRequest imagePrompt = {
+                        prompt: string `Greeting card design for ${occasion}, ${specialNotes}`
+                    };
+                    images:ImagesResponse imageRes = check openAIImages->/images/generations.post(imagePrompt);
+                    return imageRes.data[0].url;
+                }
```

Comment:
looks like the formatting is incorrect. @DevAakif please use Ballerina VSCode formatter